flink-libraries/flink-table/src/main/scala/org/apache/flink/table/expressions/ExpressionParser.scala

```diff
@@ -210,6 +212,14 @@ object ExpressionParser extends JavaTokenParsers with PackratParsers {
     case agg ~ _ ~ windowRef => UnresolvedOverCall(agg, windowRef)
   }
 
+  lazy val between: PackratParser[Expression] =
+    expression ~ "." ~ (BETWEEN | NOT_BETWEEN) ~ "(" ~ expression ~ "," ~ expression ~ ")" ^^ {
```

Comment:
The changes in `ExpressionParser` should not be necessary if you add the expression classes to `org.apache.flink.table.validate.FunctionCatalog`.  What happens if you remove these changes?